### PR TITLE
feat(sheet): group sheet get cell intercptor to performance well

### PR DESCRIPTION
**Pull request by VicKun4937**
- refactor cell inerceptor

### CODE REVIEW

**Review by wzhudev — APPROVED** [review]
LGTM

**Comment by VicKun4937** [comment]
image

It seems not solve the performance issue, because the number format is a intercepror change style and value both?

should we contiue this draft code

File: packages/core/src/sheets/view-model.ts [hunk 1]
```diff
@@ -71,7 +72,23 @@ export class SheetViewModel extends Disposable {
...
     getCell(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
         if (this._cellContentInterceptor) {
-            return this._cellContentInterceptor.getCell(row, col);
+            return this._cellContentInterceptor.getCell(row, col, InterceptorEffectEnum.Value | InterceptorEffectEnum.Style);
+        }
+
+        return this.getRawCell(row, col);
+    }
+
+    getCellOnlyValue(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
```
Comment (lumixraku) [inline packages/core/src/sheets/view-model.ts:81]: getCellValueOnly

File: packages/core/src/sheets/view-model.ts [hunk 2]
```diff
@@ -71,7 +72,23 @@ export class SheetViewModel extends Disposable {
...
+            return this._cellContentInterceptor.getCell(row, col, InterceptorEffectEnum.Value);
+        }
+
+        return this.getRawCell(row, col);
+    }
+
+    getCellOnlyStyle(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
+        if (this._cellContentInterceptor) {
+            return this._cellContentInterceptor.getCell(row, col, InterceptorEffectEnum.Style);
         }
```
Comment (lumixraku) [inline packages/core/src/sheets/view-model.ts:92]: getCellStyleOnly

File: packages/core/src/sheets/view-model.ts [hunk 3]
```diff
@@ -71,7 +72,23 @@ export class SheetViewModel extends Disposable {
...
+
+    getCellOnlyValue(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
+        if (this._cellContentInterceptor) {
+            return this._cellContentInterceptor.getCell(row, col, InterceptorEffectEnum.Value);
+        }
+
+        return this.getRawCell(row, col);
+    }
+
+    getCellOnlyStyle(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
```
Comment (lumixraku) [inline packages/core/src/sheets/view-model.ts:89]: getCellStyleOnly

File: packages/core/src/sheets/view-model.ts (hunk 1 above)
Comment (VicKun4937) [reply packages/core/src/sheets/view-model.ts:81]: ok, it will be rename next commit

**Review by Dushusir — APPROVED** [review]

File: packages/core/src/sheets/view-model.ts [hunk 4]
```diff
@@ -71,7 +72,23 @@ export class SheetViewModel extends Disposable {
...
+
+    getCellValueOnly(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
+        if (this._cellContentInterceptor) {
+            return this._cellContentInterceptor.getCell(row, col, InterceptorEffectEnum.Value);
+        }
+
+        return this.getRawCell(row, col);
+    }
+
+    getCellStyleOnly(row: number, col: number): Nullable<ICellDataForSheetInterceptor> {
```
Comment (VicKun4937) [inline packages/core/src/sheets/view-model.ts:89]: @lumixraku  here , after merged this pr, you can use above two api in sketon to get better performance

Comment (lumixraku) [reply packages/core/src/sheets/view-model.ts:89]: merge now?

**Pull request merged**